metaflow/plugins/resources_decorator.py

```diff
@@ -0,0 +1,62 @@
+from metaflow.decorators import StepDecorator
+
+
+class ResourcesDecorator(StepDecorator):
```

Comment:
I am a little concerned with this pattern where we'd just pump in more options if more platforms were added to Metaflow. Guessing we didn't go with a specific `kfp` resources decorator because that wouldn't be as simple for scientists to remember?

Reply:
True. I was hoping that a single `@resource` decorator can be used across platforms since they are similar - mostly just setting cpu, gpu and memories. You can see me also trying to make the existing default for AWS batch work for KFP use case, while also allowing more granular control by opening up more options.

I am assuming in most platforms you can either specify exact resource requirement (like AWS batch), or a range for min and max resource needed (like kfp and kubernetes). There's not much more to be added even if we have new platforms. Also this resource decorator is used more like a key value store than an actual decorator.

Reply:
The @resource decorator seems meta enough, see https://github.com/Netflix/metaflow/issues/336.  I like how the default for `cpu` is what is requested, and if the platform (Batch, Mesos, K8s, etc) supports a limit then append with `_limit` -> `cpu_limit`.

Reply:
Yeah I'm guessing it won't diverge too much so happy to defer this decision for now given the UX benefits.